### PR TITLE
MULE-9604: Add support for package and resource export on Extensions

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/util/ExtensionsTestUtils.java [hunk 1]
```diff
@@ -166,4 +171,36 @@ public static File createManifestFileIfNecessary(File targetDirectory, Manifest
...
+            @SuppressWarnings("rawtypes")
+            List differences = detDiff.getAllDifferences();
+            StringBuilder diffLines = new StringBuilder();
+            for (Object object : differences)
+            {
+                Difference difference = (Difference) object;
+                diffLines.append(difference.toString() + '\n');
+            }
+
+            throw new IllegalArgumentException(String.format("The Output for extension [%s] schema was not the expected:", diffLines.toString()));
```
Comment (elrodro83) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/util/ExtensionsTestUtils.java]: As in https://github.com/mulesoft/mule/pull/2606/files#diff-3e17b84137f7ebcbc5ea22308cb78df2L124, the extensionUnderTest name should be put there.

File: modules/extensions-support/pom.xml [hunk 2]
```diff
@@ -114,6 +114,11 @@
             <version>4.0-SNAPSHOT</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>xmlunit</groupId>
```
Comment (elrodro83) [inline modules/extensions-support/pom.xml]: Is this dependency still used in extensions-spring-support?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/ExportedArtifactsCollector.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    {
+        this.extensionModel = extensionModel;
+    }
+
+    /**
+     * @return The {@link Set} of default resource paths that the extension should export
+     */
+    Set<String> getExportedResources()
+    {
+        // TODO: remove at Kraan's notice
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/ExportedArtifactsCollector.java]: Is there a jira issue for this?

Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/ExportedArtifactsCollector.java]: not AFAIK. But already discussed this with him

File: modules/extensions-support/pom.xml (hunk 2 above)
Comment (marianogonzalez) [inline modules/extensions-support/pom.xml]: yes

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java [hunk 4]
```diff
@@ -24,13 +19,21 @@
     public static final String SPRING_FRAMEWORK_NAMESPACE = "http://www.springframework.org/schema/beans";
     public static final String SPRING_FRAMEWORK_SCHEMA_LOCATION = "http://www.springframework.org/schema/beans/spring-beans-3.0.xsd";
     public static final String MULE_SCHEMA_LOCATION = "http://www.mulesoft.org/schema/mule/core/current/mule.xsd";
+
+    public static final String MULE_EXTENSION_NAMESPACE = "http://www.mulesoft.org/schema/mule/extension";
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java]: Why not use DEFAULT_SCHEMA_LOCATION_MASK?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java [hunk 5]
```diff
@@ -24,13 +19,21 @@
...
+    public static final String MULE_EXTENSION_PREFIX = "extension";
     public static final String MULE_EXTENSION_SCHEMA_LOCATION = "http://www.mulesoft.org/schema/mule/extension/current/mule-extension.xsd";
+
     public static final String MULE_TLS_NAMESPACE = "http://www.mulesoft.org/schema/mule/tls";
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java]: Why not use DEFAULT_SCHEMA_LOCATION_MASK?

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/util/ExtensionsTestUtils.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/util/ExtensionsTestUtils.java]: actually, this method is generic and shouldn't reference any extension at all. Changed

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java (hunk 4 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/xml/SchemaConstants.java]: the mask most likely didn't exist at the time. Enhancement accepted though.

**Pull request merged**